core/src/main/java/com/microsoft/applicationinsights/internal/statsbeat/BaseStatsbeat.java

```diff
@@ -0,0 +1,82 @@
+ * substantial portions of the Software.
+ * THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR
+ * PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE
+ * FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+ * OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ */
+
+package com.microsoft.applicationinsights.internal.statsbeat;
+
+import com.microsoft.applicationinsights.TelemetryClient;
+import com.microsoft.applicationinsights.TelemetryConfiguration;
+import com.microsoft.applicationinsights.internal.util.ThreadPoolUtils;
+import com.microsoft.applicationinsights.telemetry.MetricTelemetry;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+import static com.microsoft.applicationinsights.internal.statsbeat.Constants.STATSBEAT_TELEMETRY_NAME;
+
+public abstract class BaseStatsbeat {
+
+    private static final Logger logger = LoggerFactory.getLogger(BaseStatsbeat.class);
+    protected final TelemetryClient telemetryClient;
+    protected static final ScheduledExecutorService scheduledExecutor =
+            Executors.newSingleThreadScheduledExecutor(ThreadPoolUtils.createDaemonThreadFactory(BaseStatsbeat.class));
+    protected final long interval;
+
+    public BaseStatsbeat(TelemetryClient telemetryClient, long interval) {
+        this.telemetryClient = telemetryClient;
+        this.interval = interval;
+        scheduledExecutor.scheduleAtFixedRate(sendStatsbeat(), interval, interval, TimeUnit.SECONDS);
+    }
+
+    protected abstract void send();
+
+    protected MetricTelemetry createStatsbeatTelemetry(String name, double value) {
+        MetricTelemetry telemetry = new MetricTelemetry(name, value);
+        telemetry.setTelemetryName(STATSBEAT_TELEMETRY_NAME);
+        telemetry.getContext().setInstrumentationKey(TelemetryConfiguration.getActive().getStatsbeatInstrumentationKey());
+        telemetry.getProperties().putAll(CustomDimensions.getInstance().getProperties());
+        return telemetry;
+    }
+
+    /**
+     * Runnable which is responsible for calling the send method to transmit Statsbeat telemetry
+     * @return Runnable which has logic to send statsbeat.
+     */
+    protected Runnable sendStatsbeat() {
+        return new Runnable() {
+            @Override
+            public void run() {
+                try {
+                    send();
+                }
+                catch (Exception e) {
+                    logger.error("Error occurred while sending statsbeat", e);
+                }
+            }
+        };
+    }
```

Comment:
the method name is confusing, maybe `createStatsbeatSender`? or maybe better to extract as named inner class `StatsbeatSender` and no need for method, just call `new StatsbeatSender()` above